versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/SetPropertyMojo.java

```diff
@@ -172,7 +173,14 @@ private void update(MutableXMLStreamReader pom, Property[] propertiesConfig, Str
             PropertyVersions version = entry.getValue();
             String newVersionGiven = currentProperty.getVersion();
             final String profileToApply = isEmpty(profileId) ? version.getProfileId() : profileId;
-            final String currentVersion = getProject().getProperties().getProperty(currentProperty.getName());
+            String currentVersion = getProject().getProperties().getProperty(currentProperty.getName());
+            if (currentVersion == null && isNotEmpty(profileId)) {
```

Comment:
I will try here with `profileToApply` variable.

Reply:
I refactored the condition to make the code more readable keeping the condition separated with just one `if` statement, thanks @slawekjaranowski